src/axom/multimat/multimat.hpp

```diff
@@ -487,40 +511,123 @@ class MultiMat
+
+  /*!
+   * \brief Returns the associated material set.
+   */
+  RangeSetType& getMatSet() { return m_sets[1]; }
+  const RangeSetType& getMatSet() const { return m_sets[1]; }
+
+  /*!
+   * \brief Returns a reference to the corresponding array of offsets for a
+   *        static relation corresponding to a layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  std::vector<SetPosType>& getRelBeginVec(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to the corresponding array of indices for
+   *        a static relation corresponding to a layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  std::vector<SetPosType>& getRelIndVec(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to the static relation corresponding to a
+   *        layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  StaticVariableRelationType& getRelStatic(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to the dynamic relation corresponding to a
+   *        layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  DynamicVariableRelationType& getRelDynamic(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to the dominant set of elements in the relation
+   *        corresponding to a layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  RangeSetType& getRelDominantSet(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to the secondary set of elements in the relation
+   *        corresponding to a layout.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  RangeSetType& getRelSecondarySet(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to a sparse set corresponding to a relation.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  RelationSetType& getRelSparseSet(DataLayout layout);
+
+  /*!
+   * \brief Returns a reference to a product set corresponding to a relation.
+   *
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  ProductSetType& getRelDenseSet(DataLayout layout);
+
+  /*!
+   * \brief Returns true if the static relation corresponding to the given data
+   *        layout is valid.
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  bool hasValidStaticRelation(DataLayout layout) const;
+
+  /*!
+   * \brief Returns true if the dynamic relation corresponding to the given data
+   *        layout is valid.
+   * \param layout The layout type of the relation (cell- or mat-dominant)
+   */
+  bool hasValidDynamicRelation(DataLayout layout) const;
 
 private:
   unsigned int m_ncells, m_nmats;
 
   //slam set variables
-  RangeSetType m_cellSet;
-  RangeSetType m_matSet;
+  axom::Array<RangeSetType> m_sets;
```

Comment:
Nice change! Since there are only two (I think), should this be an `axom::StackArray`? Or is it intentionally an `Array` for some other reason?

Reply:
Using `axom::Array` was intentional, since alot of Slam objects are constructed with other Slam objects as pointers -- that way, we (hopefully) just have to allocate the Slam objects in unified memory, and can defer the value-based modifications to the Slam API till later. (I think we might still need/want to go through with the latter, to avoid dealing with virtual function calls on the device)

It also allows us to just `default` our move constructors for the `MultiMat` class, since we don't have a pointer-to-instance that we need to rebind.